### ISSUE TITLE
Erroneous warning about unused Keyword return values.

### ISSUE THREAD

**Issue by holetse** [rrrene/credo#399]
### Environment

* Credo version (`mix credo -v`): 0.8.1
* Erlang/Elixir version (`elixir -v`):
Erlang/OTP 19 [erts-8.2] [source] [64-bit] [smp:8:8] [async-threads:10] [hipe] [kernel-poll:false] [dtrace]

  Elixir 1.4.1
* Operating system: OS X 10.12.5

### What were you trying to do?

Use `Keyword.get/3` with `Enum.each/2`
```
def invoke(task) when is_atom(task) do
  Enum.each(Keyword.get(MyAgent.get(:before_hooks), task, []),
    fn ([module, fnref]) -> apply(module, fnref, []) end)
  Enum.each(Keyword.get(MyAgent.get(:after_hooks), task, []),
    fn ([module, fnref]) -> apply(module, fnref, []) end)
end
```
### Expected outcome

No warnings from Credo related to unused Keyword return values.

### Actual outcome

```
  Warnings - please take a look                                                 
┃ 
┃ [W] ↗ There should be no unused return values for Keyword functions.
```

This only occurs for the first use of `Enum.each/2` and `Keyword.get/3`, and only when there is more than one. If I remove the second usage, no warning is emitted.

**Comment by philipdexter** [comment]
A few things:


The problem can be reduced to the following file

```
defmodule Myfile do

 def blah(_, _) do
  2
 end

 def invoke(task) when is_atom(task) do
   blah(Keyword.get([], :a), fn(_) -> 1 end)
   1
 end

end
```

The following file does NOT cause this issue

```
defmodule Myfile2 do

 def blah2(_) do
  1
 end

 def blah(_, _) do
  2
 end

 def invoke(task) when is_atom(task) do
   blah(Keyword.get([], :a), blah2)
   1
 end

end
```

So this issue only shows up when there is a `fn` in the arguments.

Following this thread, the issue is caused by the logic starting on line
[153 of unused_funtion_return_helper.ex](https://github.com/rrrene/credo/blob/4bb23b87555dc196a7cabda9b6c0ebea7140bd17/lib/credo/check/warning/unused_function_return_helper.ex#L153).
It seems that any fn arguments will cause the other arguments to be ignored.

If that if statement is removed and replaced with just the failure
case, then this issue is gone. However then ~7 test cases will fail.

I know too little of why the fn check is there to be able to progress
further at this time.
